src/test/java/use_cases/PlayerGameInteractorTest.java

```diff
@@ -0,0 +1,221 @@
+        game.addObserver(gameState);
+        String p1CardVal = h1.getCards().get(0).getValue();
+        String p1CardSuit = h1.getCards().get(0).getSuit();
+        p1sCard = h1.getCards().get(0);
+        interactor = new PlayerGameInteractor(game, gameState);
+        playCardReq = new PlayerGameRequestModel("sol", p1CardVal, p1CardSuit, true, false, false);
+        pickUpReq = new PlayerGameRequestModel("sol", null, null, false, true, false);
+        skipReq = new PlayerGameRequestModel("sol", null, null, false, false, true);
+
+    }
+
+    /**
+     * Set all made objects in setUp() to null to free memory.
+     */
+    @AfterEach
+    void tearDown() {
+        gameState = null;
+        p1 = null;
+        p2 = null;
+        game = null;
+        players = null;
+        interactor = null;
+        playCardReq = null;
+        skipReq = null;
+        pickUpReq = null;
+        p1sCard = null;
+        firstCard = null;
+    }
+
+    /**
+     * Test createResponse() when given the input to play a Card and the Card is valid.
+     */
+    @Test
+    public void testCreateResponsePlayValidCard() {
+        PlayerGameResponseModel response = interactor.createResponse(playCardReq);
+        // Assert turn has been changed to p3 (p2 is Computer, so logic should be done)
+        assertEquals(p3.getName(), response.getCurrentPlayerName());
+        // Assert that p1 has lost a Card and that Card is in the deck.
+        assertEquals(4, p1.getNumCards());
+        assertTrue(game.getGameDeck().getCards().contains(p1sCard));
+        // Assert the Game has no winner yet.
+        assertFalse(response.getHasWinner());
+    }
+
+    /**
+     * Test createResponse() when given the input to play a Card and the Card is invalid.
+     */
+    @Test
+    public void testCreateResponsePlayInvalidCard() {
+        // Give a new bogus Hand to p1 so to validate p1 has no valid Cards.
+        Hand newHand = new Hand(new ArrayList<>());
+        Card bogus1 = new Card("Test", "17");
+        Card bogus2 = new Card("Test", "18");
+        newHand.addCard(bogus1);
+        newHand.addCard(bogus2);
+        p1.setHand(newHand);
+        String chosenSuit = bogus1.getSuit();
+        String chosenValue = bogus1.getValue();
+        // Have p1 try to play bogus1.
+        PlayerGameRequestModel pgrm = new PlayerGameRequestModel("sol", chosenValue, chosenSuit, true, false, false);
+        interactor.createResponse(pgrm);
+        // Assert the turn has not changed, no winner, current Card is the same, none of the player's cards have changed.
+        assertEquals(p1, gameState.getCurrentPlayer());
+        assertFalse(gameState.getHasWinner());
+        assertEquals(firstCard, gameState.getCurrentCard());
+        for (Player p : game.getPlayers()) {
+            if (p.equals(p1)) {
+                assertEquals(2, p.getNumCards());
+            } else {
+                assertEquals(5, p.getNumCards());
+            }
+        }
+    }
+
```

Comment:
The tests from this point onwards are indented one too far (selecting all the lines and then shift + tab should do the trick).